src/test/regress/sql/background_task_queue_monitor.sql

```diff
@@ -279,6 +279,87 @@ SELECT job_id, task_id, status FROM pg_dist_background_task
+INSERT INTO pg_dist_background_task (job_id, command, nodes_involved) VALUES (:job_id1, $job$ SELECT pg_sleep(4); $job$, ARRAY [1, 4]) RETURNING task_id AS task_id8 \gset
+INSERT INTO pg_dist_background_task (job_id, command, nodes_involved) VALUES (:job_id1, $job$ SELECT pg_sleep(4); $job$, ARRAY [1, 3]) RETURNING task_id AS task_id9 \gset
+INSERT INTO pg_dist_background_task (job_id, command, nodes_involved) VALUES (:job_id1, $job$ SELECT pg_sleep(4); $job$, ARRAY [1, 2]) RETURNING task_id AS task_id10 \gset
+INSERT INTO pg_dist_background_task (job_id, command, nodes_involved) VALUES (:job_id1, $job$ SELECT pg_sleep(4); $job$, ARRAY [1, 2]) RETURNING task_id AS task_id11 \gset
+COMMIT;
+
+SELECT citus_task_wait(:task_id1, desired_status => 'running');
+SELECT citus_task_wait(:task_id2, desired_status => 'running');
+
+SELECT job_id, task_id, status, nodes_involved FROM pg_dist_background_task
+    WHERE task_id IN (:task_id1, :task_id2, :task_id3, :task_id4, :task_id5,
+                      :task_id6, :task_id7, :task_id8, :task_id9, :task_id10, :task_id11)
+    ORDER BY job_id, task_id; -- show that at most 1 task per node is running
+
+SELECT citus_task_wait(:task_id1, desired_status => 'done');
+SELECT citus_task_wait(:task_id2, desired_status => 'done');
+-- increase max_parallel_tasks_per_node on the fly
+ALTER SYSTEM SET citus.max_parallel_tasks_per_node = 2;
+SELECT pg_reload_conf();
+
+SELECT citus_task_wait(:task_id3, desired_status => 'running');
+SELECT citus_task_wait(:task_id4, desired_status => 'running');
+SELECT citus_task_wait(:task_id5, desired_status => 'running');
+
+SELECT job_id, task_id, status, nodes_involved FROM pg_dist_background_task
+    WHERE task_id IN (:task_id1, :task_id2, :task_id3, :task_id4, :task_id5,
+                      :task_id6, :task_id7, :task_id8, :task_id9, :task_id10, :task_id11)
+    ORDER BY job_id, task_id; -- show that at most 2 tasks per node are running
+
+-- increase to 3 max_parallel_tasks_per_node on the fly
+SELECT citus_task_wait(:task_id3, desired_status => 'done');
+SELECT citus_task_wait(:task_id4, desired_status => 'done');
+SELECT citus_task_wait(:task_id5, desired_status => 'done');
+ALTER SYSTEM SET citus.max_parallel_tasks_per_node = 3;
+SELECT pg_reload_conf();
+
+SELECT citus_task_wait(:task_id6, desired_status => 'running');
+SELECT citus_task_wait(:task_id7, desired_status => 'running');
+SELECT citus_task_wait(:task_id8, desired_status => 'running');
+
+SELECT job_id, task_id, status, nodes_involved FROM pg_dist_background_task
+    WHERE task_id IN (:task_id1, :task_id2, :task_id3, :task_id4, :task_id5,
+                      :task_id6, :task_id7, :task_id8, :task_id9, :task_id10, :task_id11)
+    ORDER BY job_id, task_id; -- show that at most 3 tasks per node are running
+
+ALTER SYSTEM RESET citus.max_parallel_tasks_per_node;
```

Comment:
One thing that's important when the limit is decreased is that if pg_cancel_backend is called on one of the running PIDs that it doesn't restart it because it's not allowed anymore by the limit. Can we test that too?